src/os/file.go

```diff
@@ -38,3 +53,136 @@ func NewFile(fd uintptr, name string) *File {
+func Open(name string) (*File, error) {
+	fd := uintptr(999)
+	switch name {
+	case "/dev/stdin":
+		fd = 0
+	case "/dev/stdout":
+		fd = 1
+	case "/dev/stderr":
+		fd = 2
+	default:
+		return nil, errors.New("unknown file: '" + name + "'")
+	}
+	return &File{fd, name}, nil
+}
+
+// OpenFile is a stub, passing through to the stub Open() call
+func OpenFile(name string, flag int, perm FileMode) (*File, error) {
+	return Open(name)
+}
+
+// Create is a stub, passing through to the stub Open() call
+func Create(name string) (*File, error) {
+	return Open(name)
+}
+
+type FileMode uint32
+
+// Mode constants
+const (
+	// The single letters are the abbreviations used by the String method's formatting.
+	ModeDir        FileMode = 1 << (32 - 1 - iota) // d: is a directory
+	ModeAppend                                     // a: append-only
+	ModeExclusive                                  // l: exclusive use
+	ModeTemporary                                  // T: temporary file; Plan 9 only
+	ModeSymlink                                    // L: symbolic link
+	ModeDevice                                     // D: device file
+	ModeNamedPipe                                  // p: named pipe (FIFO)
+	ModeSocket                                     // S: Unix domain socket
+	ModeSetuid                                     // u: setuid
+	ModeSetgid                                     // g: setgid
+	ModeCharDevice                                 // c: Unix character device, when ModeDevice is set
+	ModeSticky                                     // t: sticky
+	ModeIrregular                                  // ?: non-regular file; nothing else is known about this file
+
+	// Mask for the type bits. For regular files, none will be set.
+	ModeType = ModeDir | ModeSymlink | ModeNamedPipe | ModeSocket | ModeDevice | ModeCharDevice | ModeIrregular
+
+	ModePerm FileMode = 0777 // Unix permission bits
+)
+
+// IsDir is a stub, always returning false
+func (m FileMode) IsDir() bool {
+	return false
+}
+
+// Stub constants
+const (
+	O_RDONLY int = 1
+	O_WRONLY int = 2
+	O_RDWR   int = 4
+	O_APPEND int = 8
+	O_CREATE int = 16
+	O_EXCL   int = 32
+	O_SYNC   int = 64
+	O_TRUNC  int = 128
+)
+
+// A FileInfo describes a file and is returned by Stat and Lstat.
+type FileInfo interface {
+	Name() string       // base name of the file
+	Size() int64        // length in bytes for regular files; system-dependent for others
+	Mode() FileMode     // file mode bits
+	// ModTime() time.Time // modification time
+	IsDir() bool        // abbreviation for Mode().IsDir()
+	Sys() interface{}   // underlying data source (can return nil)
+}
+
+// Stat is a stub (for now), always returning nil
+func Stat(name string) (FileInfo, error) {
+	return nil, nil
+}
+
+// Lstat is a stub (for now), always returning nil
+func Lstat(name string) (FileInfo, error) {
+	return nil, nil
+}
+
+// Getwd is a stub (for now), always returning the string it was given
```

Comment:
> always returning the string it was given

Actually, it returns an empty string (which seems sensible to me).

Reply:
Yeah, I somehow didn't notice the brackets right before that. :wink: